### PR TITLE
add error message on PaymentType CLASS_MISMATCH_ERROR

### PR DESCRIPTION
**Pull request by FollowTheFlo**
context: Create a user on b2c, then purchase aproduct on b2b, paymentMethod step will load infinitly without an error displayed.
This code change add a global message only in this specifuc situation (when CLASS_MISMATCH_ERROR is returned from occ endpoint) and hide the spinner
-> added globalMessage at component level
-> message 'You are not authorized to perform this action. Please contact your administrator if you think this is a mistake.'

About breaking change:
constructor-> added optional GlobalMessageService in cpnt, html template is modified only when GlobalMessageService is implemented.

### CODE REVIEW

File: feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts [hunk 1]
```diff
@@ -108,7 +134,8 @@ export class CheckoutPaymentTypeComponent {
   constructor(
     protected checkoutPaymentTypeFacade: CheckoutPaymentTypeFacade,
     protected checkoutStepService: CheckoutStepService,
-    protected activatedRoute: ActivatedRoute
+    protected activatedRoute: ActivatedRoute,
+    protected globalMessageService: GlobalMessageService
```
Comment (kimhw0630) [inline feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts]: This is breaking change ... as far as I know, you need to handle it.

Comment (kimhw0630) [reply feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts]: here is example how to handle (launchDialogService is a newly added property in the constructor)
image

File: feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.html [hunk 2]
```diff
@@ -3,79 +3,83 @@ <h2 class="cx-checkout-title d-none d-lg-block d-xl-block">
...
-    </div>
-
-    <div class="row">
-      <div class="col-md-12 col-lg-6">
-        <label class="cx-payment-type-container">
-          <span class="label-content">{{
-            'paymentTypes.title' | cxTranslate
-          }}</span>
-          <div class="form-check" *ngFor="let type of paymentTypes">
+  <ng-container *ngIf="!paymentTypesError">
```
Comment (FollowTheFlo) [inline feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.html:6]: This condition is the only change made in template: don't display form if an error occurs when fetching PaymentTypes

File: feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts (hunk 1 above)
Comment (FollowTheFlo) [reply feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts]: Thanks Hak, i'll check it out LaunchDialog for an error message

**Review by kimhw0630 — APPROVED** [review]

**Pull request merged**